### PR TITLE
Add com.snowplowanalytics.mobile/diagnostic_error/jsonschema/1-0-0

### PR DESCRIPTION
**Pull request by AlexBenny**
This is the specification for a new event generated when an internal error occurs in a mobile tracker.
It can be used as diagnostic of the tracker and it can be helpful to spot in advance anomalies in the tracker's behaviour.
The diagnostic_error events can be collected by the customer and they are activated enabling a `diagnostic` feature in the mobile tracker settings.

It's related to the issue https://github.com/snowplow/snowplow-android-tracker/issues/343
https://github.com/snowplow/snowplow-android-tracker/pull/344

### CODE REVIEW

**Comment by AlexBenny** [comment]
@chuwy It's my first time pushing a schema, so, please, let me know anything missing and/or any doc reference useful for this PR.

File: schemas/com.snowplowanalytics.mobile/diagnostic_error/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "name": "diagnostic_error",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+  "type": "object",
+  "properties": {
+    "message": {
+      "description": "Message of the error.",
+      "type": "string",
+      "maxLength": 2048
```
Comment (colmsnowplow) [inline schemas/com.snowplowanalytics.mobile/diagnostic_error/jsonschema/1-0-0]: For all of these, if they can have a minimum Length it's generally preferable to have that too.

**Comment by AlexBenny** [comment]
Thanks @colmsnowplow , fixed!

**Comment by chuwy** [comment]
Hey @AlexBenny! The schema looks good, but wondering if you considered [application_error](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/application_error/jsonschema/1-0-2) as a more generic one?

**Comment by chuwy** [comment]
Closing in favor of https://github.com/snowplow/iglu-central/pull/1029

**Pull request closed without merging**